### PR TITLE
[BugFix] Fix compaction scheduler hang when tablet has many segments and disk io full

### LINKED ISSUE #21104: Compaction scheduler slow when tablet has many segments and disk io full
### Steps to reproduce the behavior (Required)
```
Thread 217 (Thread 0x7f13dc772700 (LWP 24347)):
#0  0x00007f175c448fc3 in pread64 () from /lib64/libpthread.so.0
#1  0x0000000003e9b8b3 in starrocks::io::FdInputStream::read(void*, long) ()
#2  0x0000000003e9c4b0 in starrocks::io::Readable::read_fully(void*, long) ()
#3  0x0000000003e9c00d in starrocks::io::SeekableInputStream::read_at_fully(long, void*, long) ()
#4  0x0000000003fe3c6c in starrocks::Segment::parse_segment_footer(starrocks::RandomAccessFile*, starrocks::SegmentFooterPB*, unsigned long*, starrocks::FooterPointerPB const*) ()
#5  0x0000000003fe6723 in starrocks::Segment::_open(unsigned long*, starrocks::FooterPointerPB const*) ()
#6  0x0000000003fe6b20 in starrocks::Segment::open(std::shared_ptr<starrocks::FileSystem>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned int, starrocks::TabletSchema const*, unsigned long*, starrocks::FooterPointerPB const*) ()
#7  0x000000000461a53a in starrocks::Rowset::do_load() ()
#8  0x000000000461ab4f in starrocks::Rowset::load() ()
#9  0x000000000461ad55 in starrocks::Rowset::estimate_compaction_segment_iterator_num() ()
#10 0x00000000045d0e87 in starrocks::CompactionTaskFactory::create_compaction_task() ()
#11 0x00000000040f9dff in starrocks::vectorized::SizeTieredCompactionPolicy::create_compaction(std::shared_ptr<starrocks::Tablet>) ()
#12 0x0000000003f1d402 in starrocks::Tablet::create_compaction_task() ()
#13 0x00000000040f3b8d in starrocks::CompactionManager::_try_get_next_compaction_task() ()
#14 0x00000000040f3e2e in starrocks::CompactionManager::_schedule() ()
#15 0x0000000007cabb80 in execute_native_thread_routine ()
#16 0x00007f175c441ea5 in start_thread () from /lib64/libpthread.so.0
#17 0x00007f175ba5cb0d in clone () from /lib64/libc.so.6
```

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by meegoo**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #21104

## Problem Summary(Required) ：

https://github.com/StarRocks/starrocks/pull/7225 acquire segment `num_rows` to skip empty segment, it may be hang when disk io full, so that we move create_compaction_task to thread_pool.
```
Thread 217 (Thread 0x7f13dc772700 (LWP 24347)):
#0  0x00007f175c448fc3 in pread64 () from /lib64/libpthread.so.0
#1  0x0000000003e9b8b3 in starrocks::io::FdInputStream::read(void*, long) ()
#2  0x0000000003e9c4b0 in starrocks::io::Readable::read_fully(void*, long) ()
#3  0x0000000003e9c00d in starrocks::io::SeekableInputStream::read_at_fully(long, void*, long) ()
#4  0x0000000003fe3c6c in starrocks::Segment::parse_segment_footer(starrocks::RandomAccessFile*, starrocks::SegmentFooterPB*, unsigned long*, starrocks::FooterPointerPB const*) ()
#5  0x0000000003fe6723 in starrocks::Segment::_open(unsigned long*, starrocks::FooterPointerPB const*) ()
#6  0x0000000003fe6b20 in starrocks::Segment::open(std::shared_ptr<starrocks::FileSystem>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned int, starrocks::TabletSchema const*, unsigned long*, starrocks::FooterPointerPB const*) ()
#7  0x000000000461a53a in starrocks::Rowset::do_load() ()
#8  0x000000000461ab4f in starrocks::Rowset::load() ()
#9  0x000000000461ad55 in starrocks::Rowset::estimate_compaction_segment_iterator_num() ()
#10 0x00000000045d0e87 in starrocks::CompactionTaskFactory::create_compaction_task() ()
#11 0x00000000040f9dff in starrocks::vectorized::SizeTieredCompactionPolicy::create_compaction(std::shared_ptr<starrocks::Tablet>) ()
#12 0x0000000003f1d402 in starrocks::Tablet::create_compaction_task() ()
#13 0x00000000040f3b8d in starrocks::CompactionManager::_try_get_next_compaction_task() ()
#14 0x00000000040f3e2e in starrocks::CompactionManager::_schedule() ()
#15 0x0000000007cabb80 in execute_native_thread_routine ()
#16 0x00007f175c441ea5 in start_thread () from /lib64/libpthread.so.0
#17 0x00007f175ba5cb0d in clone () from /lib64/libc.so.6
```

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by trueeyu** [comment]
Can I get this value from RocksDB？

File: be/src/storage/compaction_manager.cpp [hunk 1]
```diff
@@ -70,34 +70,40 @@ void CompactionManager::_schedule() {
...
+                      << ", compaction_type:" << starrocks::to_string(compaction_candidate.type)
+                      << ", compaction_score:" << compaction_candidate.score << " for round:" << _round
                       << ", task_queue_size:" << candidates_size();
-            auto st = _compaction_pool->submit_func([compaction_task] { compaction_task->start(); });
+            auto st = _compaction_pool->submit_func([&compaction_candidate, &task_id] {
+                auto compaction_task = compaction_candidate.tablet->create_compaction_task();
+                if (compaction_task == nullptr) {
+                    LOG(WARNING) << "create compaction task failed for tablet:" << compaction_candidate.to_string()
+                                 << ", task_id:" << task_id;
+                    return;
```
Comment (trueeyu) [inline be/src/storage/compaction_manager.cpp:97]: Why return?

Comment (meegoo) [reply be/src/storage/compaction_manager.cpp:97]: create_compaction_task return nullptr means compaction_task is running or disable.

**Comment by meegoo** [comment]
> Can I get this value from RocksDB？
seems no need, load() always needs to be called before compaction is executed.

**Review by trueeyu — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**